### PR TITLE
Config bind

### PR DESCRIPTION
**Pull request by phongsi**
this will allow people to separate out the appserver into its own container or server from the web server. Useful when there are compliance rules to separate these layers. Or if there is no benefit for a web server like when serving an api and routing on some other layer like the lb.
[issue](https://github.com/ajgon/opsworks_ruby/issues/162)

### CODE REVIEW

File: docs/source/attributes.rst [hunk 1]
```diff
@@ -430,6 +430,13 @@ appserver
     these approaches varies between appservers.  See their documentation for more
     details.
 
+- ``app['appserver']['port']``
+
+  - **Default:** None
+  - Bind the appserver to a port on 127.0.0.1.  This is
```
Comment (ajgon) [inline docs/source/attributes.rst]: Shouldn’t be `0.0.0.0` here?

File: templates/default/thin.yml.erb [hunk 2]
```diff
@@ -14,6 +14,9 @@ daemonize: true
 <% if @webserver == 'apache2' %>
 address: "127.0.0.1"
 port: 3000
+<% elsif @out[:port] %>
+address: "127.0.0.1"
+port: <%= @out[:port] %>
```
Comment (ajgon) [inline templates/default/thin.yml.erb]: Same here (as for `puma.rb`)

File: templates/default/unicorn.conf.erb [hunk 3]
```diff
@@ -5,6 +5,8 @@ user "<%= node['deployer']['user'] %>"
 working_directory "<%= @deploy_dir %>/current"
 <% if @webserver == 'apache2' %>
 listen "127.0.0.1:3000",
+<% elsif @out[:port] %>
+listen "127.0.0.1:<%= @out[:port] %>",
```
Comment (ajgon) [inline templates/default/unicorn.conf.erb]: Same here (as for `puma.rb`)

File: templates/default/puma.rb.erb [hunk 4]
```diff
@@ -56,6 +56,8 @@ threads <%= @out[:thread_min] %>, <%= @out[:thread_max] %>
 # bind 'ssl://127.0.0.1:9292?key=path_to_key&cert=path_to_cert'
 <% if @webserver == 'apache2' %>
 bind "tcp://127.0.0.1:3000"
+<% elsif @out[:port] %>
+bind "tcp://0.0.0.0:<%= @out[:port] %>"
```
Comment (ajgon) [inline templates/default/puma.rb.erb]: Can you switch conditions here, to make this webserver independent? Something like

```ruby
<% if @out[:port] %>
bind "tcp://0.0.0.0:<%= @out[:port] %>"
<% elsif @webserver == 'apache2' %>
bind "tcp://127.0.0.1:3000"
```

File: docs/source/attributes.rst (hunk 1 above)
Comment (phongsi) [reply docs/source/attributes.rst]: good catch. had to look up the difference.

File: templates/default/puma.rb.erb (hunk 4 above)
Comment (phongsi) [reply templates/default/puma.rb.erb]: great suggestion. that'll make this more flexible.

**Pull request merged**